internal/blockchain/thresholdstate.go

```diff
@@ -602,18 +575,20 @@ func (b *BlockChain) StateLastChangedHeight(hash *chainhash.Hash, version uint32
 // given deployment ID for the block AFTER the provided block hash.
 //
 // This function is safe for concurrent access.
-func (b *BlockChain) NextThresholdState(hash *chainhash.Hash, version uint32, deploymentID string) (ThresholdStateTuple, error) {
+func (b *BlockChain) NextThresholdState(hash *chainhash.Hash, deploymentID string) (ThresholdStateTuple, error) {
 	node := b.index.LookupNode(hash)
 	if node == nil || !b.index.CanValidate(node) {
-		invalidState := ThresholdStateTuple{
-			State:  ThresholdInvalid,
-			Choice: invalidChoice,
-		}
-		return invalidState, unknownBlockError(hash)
+		return invalidThresholdState, unknownBlockError(hash)
```

Comment:
I noticed this as well while going through.  It's because the function doesn't return a pointer to avoid the allocations it would otherwise cause, so `nil` isn't an option.  It could return the zero value of the struct instead, but I was trying to limit the logic changes in this PR.

I do plan to get rid of the whole invalid states (both the threshold state and the option) altogether in a future PR since it's not really needed given the error must always be checked and is, as you pointed out, not very Go like.  However, it will take a bit of time to go through and make sure no assumptions are violated.